proto/p2p/p2p.proto

```diff
@@ -349,6 +350,12 @@ message AppResponse {
   bytes app_bytes = 3;
 }
 
+message AppError {
+  bytes chain_id = 1;
+  uint32 request_id = 2;
+  string error = 3;
```

Comment:
Other considerations are to use an error code here.

Reply:
Perhaps we could consider both an error code and a string? I think there is a lot we can learn from the gRPC error model[1]. What gRPC did was generalize errors by a high level code allowing you to have a more descriptive message. I think this allows for standardization on the error code and more flexibility in the string message.

[1] https://cloud.google.com/apis/design/errors#error_model
[2] https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto#L32

Reply:
Yeah this was what i was leaning towards as posted in Slack of using an error code for matching/error handling but having the message for clarity. I think this is more robust since relying on error strings is brittle.